pkg/daemon/daemon.go

```diff
@@ -460,6 +465,18 @@ func (dn *Daemon) syncNode(key string) error {
 		return nil
 	}
 
+	// This allows skip behavior based upon the presence of
+	// /run/machine-config-daemon-force, though I don't know if that is desirable
+	// here.
+	if err := dn.validateCurrentOnDiskStateInSyncNode(); err != nil {
```

Comment:
2 thoughts:

1. this technically double validates if dn.booting, since checkStateOnFirstRun() also validates
2. this technically differs from checkStateOnFirstRun() since that forces an upgrade (even if current==desired) which this does not

Random thought: maybe we should consolidate the checkStateOnFirstRun loop? Perhaps we want to be a bit more fancy during sync checks than just this.

Reply:
We should definitely consolidate / simplify the `checkStateOnFirstRun` loop, though I don't think this PR is the best place to do that.